hack/verify-govulncheck.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+export WORKDIR=${ARTIFACTS:-$TMPDIR}
+export PATH=$PATH:$GOPATH/bin
+mkdir -p "${WORKDIR}"
+go install golang.org/x/vuln/cmd/govulncheck@v1.0.1
+
```

Comment:
use `KUBE_VERIFY_GIT_BRANCH` which is already populated in the verify CI job (other scripts in this repo already use it)

```
# KUBE_VERIFY_GIT_BRANCH is populated in verify CI jobs
BRANCH="${KUBE_VERIFY_GIT_BRANCH:-master}"
```

Reply:
`master verify all` failed for this verify script.

KUBE_VERIFY_GIT_BRANCH is overridden by Makefile.

https://github.com/kubernetes/kubernetes/blob/8319322bab8d0f135ddf9728ef795daebe81efdb/build/root/Makefile#L127-L129

We got `fatal: invalid reference: master` in

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-verify-1-29/1727556821872283648